django/db/models/fields/json.py

```diff
@@ -361,11 +361,15 @@ def as_oracle(self, compiler, connection):
         lhs, params, key_transforms = self.preprocess_lhs(compiler, connection)
         json_path = compile_json_path(key_transforms)
         if connection.features.supports_primitives_in_json_field:
-            sql = (
-                "COALESCE(JSON_VALUE(%s, '%s'), JSON_QUERY(%s, '%s' DISALLOW SCALARS))"
-            )
+            sql = """COALESCE(
+                JSON_VALUE(%s, q'\uffff%s\uffff'),
+                JSON_QUERY(%s, q'\uffff%s\uffff' DISALLOW SCALARS)
+            )"""
         else:
-            sql = "COALESCE(JSON_QUERY(%s, '%s'), JSON_VALUE(%s, '%s'))"
+            sql = """COALESCE(
+                JSON_QUERY(%s, q'\uffff%s\uffff'),
+                JSON_VALUE(%s, q'\uffff%s\uffff')
+            )"""
```

Comment:
Breaking these into multiple lines for readability and to avoid triggering flake8. The whitespaces won't be dedented though, so they will be in the query. We can avoid it by using implicit string concatenation, but I think that would make it less readable.

Reply:
Have you considered using `textwrap.dedent` to reduce the left-side whitespace? Something like:
```py
            sql = textwrap.dedent(
                """
                COALESCE(
                    JSON_VALUE(%s, q'\uffff%s\uffff'),
                    JSON_QUERY(%s, q'\uffff%s\uffff' DISALLOW SCALARS)
                )"""
            )
```

Reply:
Is that what you are referring to as implicit string concatenation?

```python
            sql = (
                "COALESCE("
                "JSON_VALUE(%s, q'\uffff%s\uffff')"
                "JSON_QUERY(%s, q'\uffff%s\uffff' DISALLOW SCALARS)"
                ")"
            )
```

I think that's a pattern we've used in a few places without readability concerns.

Reply:
@jacobtylerwalls I did, but I don't think we've used it in the codebase (I think it would incur _some_ runtime cost).

@charettes Yep! Good point about us using it in other places, I'm sure I've seen it too. I'll change it to use that.